File: implementations/julia/generate.jl
include("cid.jl")
using .CID

function main()
    ispath(CID.CIDS_DIR) || mkpath(CID.CIDS_DIR)

    for name in sort(readdir(CID.EXAMPLES_DIR))
        path = joinpath(CID.EXAMPLES_DIR, name)
        if isdir(path)
            continue
        end

        content = read(path)
        cid = CID.compute_cid(content)
        destination = joinpath(CID.CIDS_DIR, cid)
        write(destination, content)
        println("Wrote $(basename(destination)) from $name")
    end

    return 0
end

exit(main())

File: implementations/elixir/generate.exs
Code.require_file("cid.exs", __DIR__)

File.mkdir_p!(CID.cids_dir())

CID.examples_dir()
|> File.ls!()
|> Enum.map(&Path.join(CID.examples_dir(), &1))
|> Enum.reject(&File.dir?/1)
|> Enum.sort()
|> Enum.each(fn example ->
  content = File.read!(example)
  cid = CID.compute_cid(content)
  destination = Path.join(CID.cids_dir(), cid)
  File.write!(destination, content)
  IO.puts("Wrote #{Path.basename(destination)} from #{Path.basename(example)}")
end)
